### ISSUE TITLE
Environment entry behavior vs documentation

### ISSUE THREAD

**Issue by DRuggeri** [geofffranks/spruce#266]
Some time since 1.11.0 to now, the documentation at [this location](https://github.com/geofffranks/spruce/blob/master/doc/environment-variables-and-defaults.md) became incorrect (bug? feature? *shrug*).

If the ENV entry that is provided appears to be YAML, spruce injects it as YML like so:
```
$ export SOMEVAL="
root:
  child: childVal"

$ spruce merge <(echo 'result: (( grab $SOMEVAL ))')
```

The output received is:
```
result:
  root:
    child: childVal
```

But I would have expected:
```
result: |-
  root:
    child: childVal
```

Is this a bug or on purpose?

**Comment by DRuggeri** [comment]
For what it's worth, when trying to coax to a literal by using concat, I get the `go` string version of the `map[string]interface{}` instead:
```
$ spruce merge <(echo 'result: (( concat $SOMEVAL " " ))')
result: 'map[root:map[child:childVal]] '
```

**Comment by DRuggeri** [comment]
Another clarification. Seems to be caused by 1.12.2 with release note of:
```
When pulling environment variables in, their values are now
unmarshaled into appropriate types via YAML. For example, PORT=123
will be used as an integer in the output YAML. This may cause problems
if a consumer of the output YAML is requiring a string value of the number/boolean
being displayed. If this is the case, (( concat $PORT "" )) can be
used as a workaround. It seems more likely that there will be consumers
requiring the value as an integer/boolean type rather the stringified version.
```

So, this is "kinda" expected. It seems the unmarshalling code is going way beyond ints/strings.

**Comment by geofffranks** [comment]
I'm going to call this a bug unless I find tests that indicate otherwise. Unmarshaling should have been limited to just scalar types, based on the context of the PR that enabled it

**Comment by librett0** [comment]
I'm not opening a new issue (unless you want me to) since this is essentially the same bug, but I'd like to point out an annoying behavior when trying to use `concat` to coerce values into strings.

If the number is a decimal, any trailing zeroes are truncated _before_ it's turned into a string, so in cases like this:

```
export VERSION=1.20

---
version: (( concat $VERSION "" ))
```

you get `version: "1.2"` instead.

**Comment by dikderoy** [comment]
I have a similar issue caused by this behavior:

I was not aware (until I looked into the code), what spruce uses yaml.Unmarshall on env variables.

I tried to `(( grab $COMMIT_SHORT_SHA ))` which was working fine until I hit this value: `79535e52`, which is treated as "math notation for floats" and parsed as float = `79535+e52` which effectively results in `.inf` in YAML and wrecks havoc.

Using `(( concat $COMMIT_SHORT_SHA "" ))` is not able to fix things here and results in ` 7.9535e+56` in output yaml.

I would love to see some option to tweak typecasting/parsing somehow. maybe a "special prefix" to the EnvVar name in the op like so:

```
(( grab !!str$ENV_VAR_NAME ))
```
